contributingGuides/TS_STYLE.md

```diff
@@ -0,0 +1,517 @@
+
+  ```ts
+  // Array<T>
+  const a: Array<string | number> = ["a", "b"];
+  const b: Array<{ prop: string }> = [{ prop: "a" }];
+  const c: Array<() => void> = [() => {}];
+
+  // T[]
+  const d: MyType[] = ["a", "b"];
+  const e: string[] = ["a", "b"];
+  const f: readonly string[] = ["a", "b"];
+  ```
+
+<a name="ts-ignore"></a><a name="1.7"></a>
+
+- [1.7](#ts-ignore) **@ts-ignore**: Do not use `@ts-ignore` or its variant `@ts-nocheck` to suppress warnings and errors.
+
+  > Use `@ts-expect-error` during the migration for type errors that should be handled later. Refer to the [Migration Guidelines](#migration-guidelines) for specific instructions on how to deal with type errors during the migration. eslint: [`@typescript-eslint/ban-ts-comment`](https://typescript-eslint.io/rules/ban-ts-comment/)
+
+<a name="ts-nullish-coalescing"></a><a name="1.8"></a>
+
+- [1.8](#ts-nullish-coalescing) **Optional chaining and nullish coalescing**: Use optional chaining and nullish coalescing instead of the `get` lodash function. eslint: [`no-restricted-imports`](https://eslint.org/docs/latest/rules/no-restricted-imports)
+
+  ```ts
+  // BAD
+  import lodashGet from "lodash/get";
+  const name = lodashGet(user, "name", "default name");
+
+  // GOOD
+  const name = user?.name ?? "default name";
+  ```
+
+<a name="type-inference"></a><a name="1.9"></a>
+
+- [1.9](#type-inference) **Type Inference**: When possible, allow the compiler to infer type of variables.
+
+  ```ts
+  // BAD
+  const foo: string = "foo";
+  const [counter, setCounter] = useState<number>(0);
+
+  // GOOD
+  const foo = "foo";
+  const [counter, setCounter] = useState(0);
+  const [username, setUsername] = useState<string | undefined>(undefined); // Username is a union type of string and undefined, and its type cannot be inferred from the default value of undefined
+  ```
+
+  For function return types, default to always typing them unless a function is simple enough to reason about its return type.
+
+  > Why? Explicit return type helps catch errors when implementation of the function changes. It also makes it easy to read code even when TypeScript intellisense is not provided.
+
+  ```ts
+  function simpleFunction(name: string) {
+    return `hello, ${name}`;
+  }
+
+  function complicatedFunction(name: string): boolean {
+    // ... some complex logic here ...
+    return foo;
+  }
+  ```
+
+<a name="jsdoc"></a><a name="1.10"></a>
+
+- [1.10](#jsdoc) **JSDoc**: Omit comments that are redundant with TypeScript. Do not declare types in `@param` or `@return` blocks. Do not write `@implements`, `@enum`, `@private`, `@override`. eslint: [`jsdoc/no-types`](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.README/rules/no-types.md)
+
+  ```ts
+  // BAD
+  /**
+   * @param {number} age
+   * @returns {boolean} Whether the person is a legal drinking age or nots
+   */
+  function canDrink(age: number): boolean {
+    return age >= 21;
+  }
+
+  // GOOD
+  /**
+   * @param age
```

Comment:
I think we should clarify that `@param` and `@returns` are optional, and only included when there's a comment that explains the parameter or return type more clearly:

```ts
// BAD

/**
 * @param {number} age
 * @returns {boolean}
 */
function canDrink(age: number): boolean {
    return age >= 21;
}

/**
 * @param age
 */
function canDrink(age: number): boolean {
    return age >= 21;
}

// GOOD

/**
 * @returns – whether the person is of legal drinking age in the United States
 */
function canDrink(age: number): boolean {
    return age >= 21;
}
```